tests/js/mock-use-instance-id.js

```diff
@@ -17,50 +17,77 @@
  */
 
 /**
- * External dependencies
+ * WordPress dependencies
  */
-import faker from 'faker';
+import { useMemoOne } from 'use-memo-one';
+import { useInstanceId } from '@wordpress/compose';
 
 /**
- * WordPress dependencies
+ * Creates a new ID for a given object.
+ *
+ * This uses a `Map` to track object instances, as opposed to the `WeakMap` used in the original `useInstanceId()` from @wordpress/compose.
+ * Using a regular `Map` to track objects will ensure they are not unpredictably garbage collected during tests, which can cause the
+ * generated instance IDs to change, with resulting test failures.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Map}    instanceMap Map of object instances to their current id.
+ * @param {Object} object      Object reference to create an id for.
+ * @return {number} The new id.
  */
-import { useMemo } from '@wordpress/element';
-import { useInstanceId } from '@wordpress/compose';
+function createID( instanceMap, object ) {
+	const instances = instanceMap.get( object ) || 0;
+	instanceMap.set( object, instances + 1 );
+
+	return instances;
+}
 
 /**
- * Provides a unique, memoized ID.
+ * Provides a unique instance ID.
  *
  * @since 1.107.0
+ * @since n.e.x.t Updated to provide instance ID instead of memoized ID.
  *
- * @param {Object} object Object reference to create an ID for. This is unused and only included for compatibility with the original `useInstanceId()`.
- * @param {string} prefix Prefix for the unique ID.
- * @return {string} The unique ID.
+ * @param {Map}    instanceMap Map of object instances to their current id.
+ * @param {Object} object      Object reference to create an id for.
+ * @param {string} prefix      Prefix for the unique id.
+ * @return {string} The unique id.
  */
-function useMemoizedID( object, prefix ) {
-	return useMemo( () => {
-		const id = faker.datatype.uuid();
+function useInstanceID( instanceMap, object, prefix = '' ) {
+	return useMemoOne( () => {
+		const id = createID( instanceMap, object );
+
 		return prefix ? `${ prefix }-${ id }` : id;
-	}, [ prefix ] );
+	}, [ instanceMap, object, prefix ] );
 }
 
+// eslint-disable-next-line sitekit/jsdoc-tag-grouping
```

Comment:
@tofumatt pretty unrealistic but had to bypass custom eslint rule to be able to commit totally valid inline doc

Reply:
Hmm, that's odd. I'll check into why that is, seems wrong 😅

Reply:
Turns out it's a bug in the ESLint code that we've yet to run into 😆 

I fixed it in the rule, to account for multiple `@since` lines at the end of a JSDoc block 🙂